base_delivery_carrier_label/models/delivery_carrier.py

```diff
@@ -38,7 +37,9 @@ def send_shipping(self, pickings):
         # or at the contrary, we could call a common method for multiple delivery_type
         # for instance, in delivery_roulier, the same method is always called for any
         # carrier implemented in roulier lib.
-        if result is None:
+        if pickings.carrier_id:
+            result = super().send_shipping(pickings)
+        else:
```

Comment:
I think this is better as `super().send_shipping(pickings)` may return None.
```suggestion
        if pickings.carrier_id:
            result = super().send_shipping(pickings)
        if result is None:
```